### DISCUSSION TITLE
Nuxt 3 - Any idea how to export to a single file?

### DISCUSSION THREAD

**Discussion by scolastico** [nuxt/nuxt#15901] [Questions] [answered]
Yes i know single files are bad but in some cases like mine there is no better workaround than a single file. Its a bit complicated to explain why but in short i  need to render the app from a single html file because the software rendering it has trouble accessing the js from the harddrive because of cors.

So is there a way to export only one js files or to change the loading so any js files are loaded from html instead of js?

I allready tried
 - https://www.npmjs.com/package/html-webpack-inline-source-plugin
 - https://www.npmjs.com/package/vite-plugin-singlefile

But both seem not to work with the nuxt structure.

**Comment by Flanker72** [comment, marked as answer]
Do you **really** need Nuxt for this? You can build plain Vue3 project using Vite with `build.lib` config and achieve the desired result.

You may try to override build process using [Nuxt build hooks](https://github.com/nuxt/framework/blob/main/packages/schema/src/types/hooks.ts#L55) and try to modify/bundle the result by yourself.

**Reply by scolastico** [reply]
I love the most auto importing features from nuxt. So i think i will try to make some kind of module which uses the build hooks. I was in hope somebody has done allready something simmilar or eventually exactly what i need.

**Reply by scolastico** [reply]
I have now written a small module (which isnt perfekt and npm ready, but could be usefull).
 
Before using install the following package: `inline-scripts` (i used version `1.7.4`)

The module: [modules.zip](https://github.com/nuxt/framework/files/8923779/modules.zip)

**Reply by jaonoctus** [reply]
@scolastico I tried to use your script but got the following error:

```
 ERROR  Cannot read properties of undefined (reading 'loaders')                                                                                                                            10:25:49

  at build:before (modules/single-file-export/index.ts:92:24)
  at node_modules/hookable/dist/index.mjs:39:82
  at async build (node_modules/nuxt/dist/index.mjs:2233:3)
  at async Object.invoke (node_modules/nuxi/dist/chunks/build.mjs:56:5)
  at async Object.invoke (node_modules/nuxi/dist/chunks/generate.mjs:32:5)
  at async _main (node_modules/nuxi/dist/cli.mjs:50:20)
  ```
